packages/fast-components-react-msft/src/progress/progress.tsx

```diff
@@ -47,48 +48,123 @@ class Progress extends Foundation<ProgressHandledProps, ProgressUnhandledProps,
                 minValue={this.props.minValue}
                 maxValue={this.props.maxValue}
             >
-                <div
-                    className={this.progressIndicatorClasses()}
-                    slot={ProgressType.determinate}
-                >
-                    <div
-                        className={get(
-                            this.props,
-                            "managedClasses.progress_valueIndicator"
-                        )}
-                        style={{ width: `${this.props.value}%` }}
-                    />
-                </div>
-                <div
-                    slot={ProgressType.indeterminate}
-                    className={get(this.props, "managedClasses.progress_indicator")}
-                >
-                    {this.renderIndeterminateItems()}
-                </div>
+                {this.renderProgress()}
             </BaseProgress>
         );
     }
 
+    /**
+     * Create class names
+     */
+    protected generateClassNames(): string {
+        let className: string = get(this.props.managedClasses, "progress", "");
+
+        if (this.props.circular) {
+            className = get(this.props.managedClasses, "progress_circular", "");
+        }
+
+        return super.generateClassNames(className);
+    }
+
+    private generateCircularValueIndicatorClassNames(): string {
+        let className: string = get(
+            this.props.managedClasses,
+            "progress_valueIndicator",
+            ""
+        );
+
+        if (!this.props.value) {
+            className = `${className} ${get(
+                this.props.managedClasses,
+                "progress_valueIndicator__indeterminate",
+                ""
+            )}`;
+        }
+
+        return className;
+    }
+
+    private generateSVGClassNames(): string {
+        const className: string = this.props.size
+            ? get(this.props.managedClasses, `progress_circular__${this.props.size}`, "")
+            : get(this.props.managedClasses, "progress_circular__container", "");
+
+        return className;
+    }
+
     private progressIndicatorClasses(): string {
         return [
-            get(this.props, "managedClasses.progress_indicator"),
-            get(this.props, "managedClasses.progress_indicator__determinate"),
+            get(this.props.managedClasses, "progress_indicator"),
+            get(this.props.managedClasses, "progress_indicator__determinate"),

```

Comment:
nit: I'd prefer we refrain from updating these in the future, it's all personal preference and doesn't actually fix anything

Reply:
I'm just matching what was done elsewhere in the file